### PR TITLE
Merge reuse into tests workflow

### PR DESCRIPTION
**Pull request by tpadioleau**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -18,6 +18,13 @@ jobs:
         extensions: 'hpp,cpp'
         clangFormatVersion: 12
 
+  reuse:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: REUSE Compliance Check
```
Comment (jbigot) [inline .github/workflows/tests.yml:25]: I'd make it a step of the lint job

```suggestion
    - uses: actions/checkout@v4
      name: REUSE Compliance Check
```

**Review by jbigot — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jbigot) [reply .github/workflows/tests.yml:25]:
```suggestion
    - name: REUSE Compliance Check
      if: always()
```
That way, it will

Comment (tpadioleau) [reply .github/workflows/tests.yml:25]: Ok why not, why is it better ?

Comment (jbigot) [reply .github/workflows/tests.yml]: We don't spin up a whole new VM for a 1s job. Should slightly decrease pressure on runners

**Pull request merged**